pkg/api/context_apis.go

```diff
@@ -0,0 +1,103 @@
+package api
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/qlik-oss/k-apis/pkg/config"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	QliksenseConfigApiVersion  = "config.qlik.com/v1"
+	QliksenseConfigKind        = "QliksenseConfig"
+	QliksenseContextApiVersion = "qlik.com/v1"
+	QliksenseContextKind       = "Qliksense"
+	QliksenseDefaultProfile    = "docker-desktop"
+	DefaultRotateKeys          = "yes"
+	QliksenseMetadataName      = "QliksenseConfigMetadata"
+)
+
+// AddCommonConfig adds common configs into CRs
+func (qliksenseCR *QliksenseCR) AddCommonConfig(contextName string) {
+	qliksenseCR.ApiVersion = QliksenseContextApiVersion
+	qliksenseCR.Kind = QliksenseContextKind
+	if qliksenseCR.Metadata == nil {
+		qliksenseCR.Metadata = &Metadata{}
+	}
+	if qliksenseCR.Metadata.Name == "" {
+		qliksenseCR.Metadata.Name = contextName
+	}
+	qliksenseCR.Spec = &config.CRSpec{}
+	qliksenseCR.Spec.Profile = QliksenseDefaultProfile
+	qliksenseCR.Spec.ReleaseName = contextName
+	qliksenseCR.Spec.RotateKeys = DefaultRotateKeys
+}
+
+// AddBaseQliksenseConfigs adds configs into config.yaml
+func (qliksenseConfig *QliksenseConfig) AddBaseQliksenseConfigs(defaultQliksenseContext string) {
+	qliksenseConfig.ApiVersion = QliksenseConfigApiVersion
+	qliksenseConfig.Kind = QliksenseConfigKind
+	if qliksenseConfig.Metadata == nil {
+		qliksenseConfig.Metadata = &Metadata{}
+	}
+	qliksenseConfig.Metadata.Name = QliksenseMetadataName
+	if defaultQliksenseContext != "" {
+		if qliksenseConfig.Spec == nil {
+			qliksenseConfig.Spec = &ContextSpec{}
+		}
+		qliksenseConfig.Spec.CurrentContext = defaultQliksenseContext
+	}
+}
+
+// WriteToFile (QliksenseCR, QliksenseConfig) writes content into specified file
+func WriteToFile(content interface{}, targetFile string) error {
```

Comment:
Just a thought, since you were using `ioutil.ReadFile()` in the `ReadFromFile()`, you might want to consider `ioutil.WriteFile()` in the `WriteToFile()`

Reply:
done